cmd/plan.go

```diff
@@ -162,7 +169,72 @@ func planInstall(rv cran.RVersion) (*cran.PkgDb, gpsr.InstallPlan) {
 	for pkg := range ip.DepDb {
 		pkgs = append(pkgs, pkg)
 	}
+
 	log.Infoln("total packages required:", len(ip.StartingPackages)+len(ip.DepDb))
 	log.Infoln("resolution time", time.Since(startTime))
 	return cdb, ip
 }
+
+
+type InstalledPackage struct {
+	Name string
+	Version string
+	Repo string
+}
+
+func GetPriorInstalledPackages(fileSystem afero.Fs, libraryPath string) map[string]InstalledPackage {
+
+	installed := make(map[string]InstalledPackage)
+
+	installedLibrary, err := fileSystem.Open(libraryPath)
+
+	if err != nil {
+		//panic?
+		return installed
+	}
+
+	installedPkgs, _ := installedLibrary.Readdir(0)
+
+	for _, f := range installedPkgs {
+		descriptionFilePath := filepath.Join(libraryPath, f.Name(), "DESCRIPTION")
+		descriptionFile, err := fileSystem.Open(descriptionFilePath)//, _ := fs.Open()
+
+		if err != nil {
+			//panic?
+			log.WithField("file", descriptionFilePath).Warn("DESCRIPTION missing from installed package")
+		}
+		scanner := bufio.NewScanner(descriptionFile)//#bufio.Scanner(fs.Open(f))
+		log.WithField("description file", descriptionFilePath).Debug("scanning DESCRIPTION file")
+
+		var pkgName, pkgVersion, pkgRepo = "", "", ""
```

Comment:
empty string is the default value for these so can just initialize as type string